app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofiles.utils
+
+import play.api.libs.json.{JsPath, JsValue, Json, JsonValidationError}
+import play.api.mvc.{Result, Results}
+import uk.gov.hmrc.tradergoodsprofiles.models.errors.{Error, InvalidErrorResponse}
+import uk.gov.hmrc.tradergoodsprofiles.models.requests.APICreateRecordRequest
+import uk.gov.hmrc.tradergoodsprofiles.services.UuidService
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ValidationSupport {
+
+  private val fieldsToErrorCode: Map[String, (String, String, Int)] = Map(
+    "/actorId"                  -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidActorMessage, ApplicationConstants.InvalidActorId),
+    "/traderRef"                -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidOrMissingTraderRef, ApplicationConstants.InvalidOrMissingTraderRefCode),
+    "/comcode"                  -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidOrMissingComcode, ApplicationConstants.InvalidOrMissingComcodeCode),
+    "/goodsDescription"         -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidOrMissingGoodsDescription, ApplicationConstants.InvalidOrMissingGoodsDescriptionCode),
+    "/countryOfOrigin"          -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidOrMissingCountryOfOrigin, ApplicationConstants.InvalidOrMissingCountryOfOriginCode),
+    "/category"                 -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidOrMissingCategory, ApplicationConstants.InvalidOrMissingCategoryCode),
+    "/comcodeEffectiveFromDate" -> (ApplicationConstants.InvalidRequestParameter, ApplicationConstants.InvalidOrMissingComcodeEffectiveFromDate, ApplicationConstants.InvalidOrMissingComcodeEffectiveFromDateCode)
+  )
+
+  def validateCreateRecordRequest(json: JsValue, uuidService: UuidService)(implicit
```

Comment:
can we move this to the CReateRecordController as this is the only place where we use it.

Reply:
one sec hold that thought for ammoment

Reply:
yes, I think this should be in the CreateRecordController as it is only use in there and is not generic

Reply:
this will be used by update endpoint eventually so its better to have it in validation support and may be rename to appropriate method as part of update endpoint

Reply:
Yes, ok, a more generic name would be fine. But again we are making the mistake of looking into the feature. The update record is already being done by Paul and Paul may uses different validation, so we may never have the time to refactor or change. So would have been better just to think about the create and then refactor when it is needed it.

Reply:
renamed the method name to be generic and Paul would be reusing this for update endpoint. thanks